src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -46,6 +45,53 @@
 SRC_POSITION_TEST_MODE = False
 
 
+class ExtractAssignmentVisitor(ast.NodeVisitor):
+    def __init__(self) -> None:
+        super().__init__()
+        self.symbols: Optional[Union[str, List[str]]] = None
+
+    def visit_Assign(self, node) -> None:
```

Comment:
I think this can be `visit_assign`, or `visit_assign_node`. The current capitalization seems unidiomatic for Python and is, frankly, more distracting to read. (`VisitAssign` or `visit_assign`)

Reply:
Unfortunately that won't work because this here is overriding a base-class method from `ast.NodeVisitor`. The naming indeed is weird, and an oddity of python. For context, the pattern is
`visit_<classname>`. https://docs.python.org/3/library/ast.html#ast.NodeVisitor.visit

Reply:
If `ast.NodeVisitor` follows a know Python idiom that violates the more common naming convention, a comment isn't strictly necessary, but having one in `ast.NodeVisitor` itself might help.

Reply:
`ast.NodeVisitor` is a builtin class. It comes as part of the python standard library :). Ok, in this I'll leave as is.

Reply:
Oh, it's not our `ast` 😄  OK, that makes sense.